src/components/account/edit/Immatriculation.vue

```diff
@@ -0,0 +1,162 @@
+        </div>
+    </div>
+</template>
+
+<script lang="ts">
+    import MultiSelect from "@/components/MultiSelect.vue";
+    import MatriculationManagement from "@/api/MatriculationManagement";
+    import { onBeforeMount, ref, computed, reactive } from "vue";
+    import { FieldOfStudy } from "@/api/api_models/user_management/FieldOfStudy";
+    import { historyToSortedList } from "@/use/ImmatriculationHistoryHandler";
+    import MatriculationData from "@/api/api_models/matriculation_management/MatriculationData";
+    import SubjectMatriculation from "@/api/api_models/matriculation_management/SubjectMatriculation";
+    import GenericResponseHandler from "@/use/GenericResponseHandler";
+    import ImmatriculationHistoryEntry from "@/components/ImmatriculationHistoryEntry.vue";
+    import LoadingSpinner from "@/components/loading/Spinner.vue";
+    import ImmatriculationHistory from "@/components/ImmatriculationHistory.vue";
+
+    export default {
+        components: {
+            MultiSelect,
+            LoadingSpinner,
+            ImmatriculationHistory,
+        },
+        props: {
+            username: {
+                type: String,
+                required: true,
+            },
+        },
+        setup(props: any) {
+            let refreshKey = ref(false);
+            let busy = ref(true);
+            let fieldsOfStudy = Object.values(FieldOfStudy).filter((e) => e != FieldOfStudy.NONE);
+            let semesterType = ref("");
+            let year = ref("");
+            let selectedFieldsOfStudy = ref([] as FieldOfStudy[]);
+
+            let currentYear = new Date().getFullYear();
+            let selectableYears = computed(() => {
+                let array: String[] = [];
+                for (let index = currentYear; index >= currentYear - 10; index--) {
+                    if (semesterType.value == "WS") {
+                        array.push(
+                            index.toString() + "/" + (index + 1).toString().substring(index.toString().length - 2, index.toString().length)
+                        );
+                    } else {
+                        array.push(index.toString());
+                    }
+                }
+                return array;
+            });
+
+            function updateSelectedFieldsOfStudy(value: any) {
+                selectedFieldsOfStudy.value = value.value;
+            }
+
+            let selectedSemester = computed(() => {
+                return semesterType.value + year.value;
+            });
+
+            let validSelection = computed(() => {
+                return (
+                    semesterType.value != "" &&
+                    year.value != "" &&
+                    selectedFieldsOfStudy.value.length != 0 &&
+                    selectedFieldsOfStudy.value[0] != ""
+                );
+            });
+
+            function resetYear() {
+                year.value = "";
+            }
+
+            async function updateImmatriculation() {
```

Comment:
I had great difficulties understanding the code in this function.
I understand that this will be easier, once the new API code is implemented, however:

I think it would greatly help if you use a for loop without indices (for each), then you would not need to have a list of indices of which updates were sucessful or not, this is very confusing imo